src/test/java/org/opentripplanner/middleware/connecteddataplatform/ConnectedDataPlatformTest.java

```diff
@@ -403,4 +406,90 @@ void shouldProcessTripHistory() {
         assertFalse(ConnectedDataManager.shouldProcessTripHistory(null));
         assertFalse(ConnectedDataManager.shouldProcessTripHistory(""));
     }
+
+    @Test
+    void canHandleMissingPlaceCoordinates() throws Exception {
+        assumeTrue(IS_END_TO_END);
+
+        String userId = UUID.randomUUID().toString();
+        String batchId = "missing-coords";
+        String mode = "WALK,BUS,RAIL,TRAM";
+        TripRequest tripRequestOne = PersistenceTestUtils.createTripRequest(
+            userId,
+            batchId,
+            DateTimeUtils.convertToDate(PREVIOUS_WHOLE_HOUR_FROM_NOW),
+            mode
+        );
+
+        // Remove coordinates from trip request and update.
+        tripRequestOne.fromPlace = "Airport, Stansted, Essex, England :: ";
+        tripRequestOne.toPlace = "Airport, Glasgow Airport, Glasgow, Scotland :: ";
+        Persistence.tripRequests.replace(tripRequestOne.id, tripRequestOne);
+        tripRequests.clear();
+        tripRequests.add(tripRequestOne);
+
+        OtpResponse planResponse = OtpTestUtils.OTP_DISPATCHER_PLAN_RESPONSE.getResponse();
+        for (Itinerary itinerary : planResponse.plan.itineraries) {
+            for (Leg leg : itinerary.legs) {
+                // Set all legs to transit so that the coordinates are extracted.
+                leg.transitLeg = true;
+            }
+        }
+        tripSummary = new TripSummary(planResponse.plan, planResponse.error, tripRequestOne.id, batchId);
+        Persistence.tripSummaries.create(tripSummary);
+
+        TripHistoryUploadJob.stageUploadHours();
+        TripHistoryUploadJob.processTripHistory(true);
+        zipFileName = getFileName(PREVIOUS_WHOLE_HOUR_FROM_NOW, ConnectedDataManager.ZIP_FILE_NAME_SUFFIX);
+        tempFile = String.join(
+            "/",
+            FileUtils.getTempDirectory().getAbsolutePath(),
+            zipFileName
+        );
+        String fileContents = getContentsOfFileInZip(
+            tempFile,
+            getFileName(PREVIOUS_WHOLE_HOUR_FROM_NOW, ConnectedDataManager.DATA_FILE_NAME_SUFFIX)
+        );
+        List<AnonymizedTripRequest> anonymizedTripRequests = JsonUtils.getPOJOFromJSONAsList(fileContents, AnonymizedTripRequest.class);
+        assertNotNull(anonymizedTripRequests);
+        // Confirm that all missing lat/lon's have the default value of 0.
+        assertEquals(0, anonymizedTripRequests.get(0).fromPlace.lat);
```

Comment:
You could have also used something like `assertEquals(zeroCoordinates, fromPlace)`.